pandas/core/index.py

```diff
@@ -508,24 +515,26 @@ def best_len(values):
             else:
                 return 0
 
+        data = self._formatter_data
```

Comment:
make a nicer ``_formatter_func``, this is what the datetimelikes do, this would override the use of ``com.pprint_thing`` (which is the default), and do a specific formatter